src/client/components/input/partial-date.js

```diff
@@ -25,7 +25,19 @@ const ymdRegex = /^\d{4}-\d{2}-\d{2}$/;
 const ymRegex = /^\d{4}-\d{2}$/;
 const yRegex = /^\d{4}$/;
 
+/**
+* This class is derived from the React Component base class
+* to render the partial date of any of the formats "YYYY", "YYYY-MM",
+* "YYYY-MM-DD" which allows the editor to be vague where information is not
+* known. It also checks validity of the value filled in the date field.
+*/
 class PartialDate extends React.Component {
+	/**
+	* Function to check validity of the date filled in the field.
+	* @param {string} value - Date filled in field is passed in the validator.
+	* @returns {boolean} - Returns true either if no data is filled in the field
```

Comment:
No need to repeat "Returns" here - just "True if the date is valid or if no date is present" is good